src/portal/lib/src/create-edit-rule/create-edit-rule.component.ts

```diff
@@ -396,23 +400,29 @@ export class CreateEditRuleComponent implements OnInit, OnDestroy {
     if (this.targetList.length === 0) {
       this.noEndpointInfo = "REPLICATION.NO_ENDPOINT_INFO";
     }
-    let registryObs = this.repService.getRegistryInfo(0);
     if (ruleId) {
       this.policyId = +ruleId;
       this.headerTitle = "REPLICATION.EDIT_POLICY_TITLE";
-      zip(registryObs, this.repService.getReplicationRule(ruleId))
-        .subscribe(([adapter, ruleInfo]) => {
-          this.setFilterAndTrigger(adapter);
-          this.copyUpdateForm = clone(ruleInfo);
-          // set filter value is [] if callback filter value is null.
-          this.updateForm(ruleInfo);
-          // keep trigger same value
-          this.copyUpdateForm.trigger = clone(ruleInfo.trigger);
-          this.copyUpdateForm.filters = this.copyUpdateForm.filters === null ? [] : this.copyUpdateForm.filters;
+      this.repService.getReplicationRule(ruleId)
+        .subscribe((ruleInfo) => {
+          let srcRegistryId = ruleInfo.src_registry.id;
+          this.repService.getRegistryInfo(srcRegistryId)
+            .subscribe(adapter => {
+              this.setFilterAndTrigger(adapter);
+              this.copyUpdateForm = clone(ruleInfo);
+              // set filter value is [] if callback filter value is null.
+              this.updateForm(ruleInfo);
+              // keep trigger same value
+              this.copyUpdateForm.trigger = clone(ruleInfo.trigger);
```

Comment:
should moved to in the front of updateform

Reply:
It has been modified